action/protocol/execution/protocol.go

```diff
@@ -83,6 +84,25 @@ func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.St
 	return receipt, nil
 }
 
+// HandleCrossProtocol handles an execution from another protocol
+func (p *Protocol) HandleCrossProtocol(ctx context.Context, act action.Action, sm protocol.StateManager) (*action.Receipt, error) {
```

Comment:
We may have a function for this protocol only as
```func (p *Protocol) HanleExecution(ctx context.Context, exec *action.Execution, sm protocol.StateManager) (*action.Receipt, error) {
    return p.handle(ctx, exec, sm)
}
```
and this function could be reused by `Handle`